datalad/ui/__init__.py

```diff
@@ -81,7 +82,7 @@ def set_backend(self, backend):
                 else:
                     backend = 'dialog'
             else:
-                backend = 'dialog' if is_interactive() else 'no-progress'
+                backend = 'dialog' if is_interactive() else 'quiet'
```

Comment:
In terms of non-interactive behavior, I agree that showing the translating a progress bar to a log message upon completion is better than silence.  The label "quiet" doesn't feel quite intuitive to me, but I guess it makes sense given the name `QuietConsoleLog`, and I don't have a better suggestion.